tests/core/mountpoint/test_base.py

```diff
@@ -674,3 +674,100 @@ async def _in_trio_land(user_fs, mountpoint_manager):
+
+    await alice_user_fs.sync()
+
+    # Add some files
+    for w in workspaces:
+        await alice_user_fs.get_workspace(workspaces[w]).touch(f"/{w}.txt")
+
+    personal_base_mountpoint = base_mountpoint / "Personal"
+    standard_base_mountpoint = base_mountpoint / "Standard"
+    async with mountpoint_manager_factory(
+        alice_user_fs,
+        alice_user_fs.event_bus,
+        standard_base_mountpoint,
+        mountpoint_in_directory=True,
+        personal_workspace_base_path=personal_base_mountpoint,
+        personal_workspace_name_pattern=personal_workspace_name_pattern,
+    ) as mountpoint_manager:
+        # Standard workspace should be mounted on standard_base_mountpoint
+        for name in standard_workspaces:
+            standard_mountpoint_path = await mountpoint_manager.mount_workspace(workspaces[name])
+            assert standard_mountpoint_path == (standard_base_mountpoint / name).absolute()
+            assert await trio.Path(standard_mountpoint_path / f"{name}.txt").exists()
+
+        # Personal workspace should be mounted on personal_base_mountpoint
+        for name in personal_workspaces:
+            personal_mountpoint_path = await mountpoint_manager.mount_workspace(workspaces[name])
+            assert personal_mountpoint_path == (personal_base_mountpoint / name).absolute()
+            assert await trio.Path(personal_mountpoint_path / f"{name}.txt").exists()
+
+
+@pytest.mark.trio
+@pytest.mark.mountpoint
+async def test_personal_workspace_invalid_options(base_mountpoint, running_backend, alice_user_fs):
+    standard_workspace_name = "standard_workspace"
+    personal_workspace_name = "personal_workspace"
+    standard_base_mountpoint = base_mountpoint / "Standard"
+    personal_base_mountpoint = base_mountpoint / "Personal"
+
+    # Create standard and personal workspaces
+    wid_standard = await alice_user_fs.workspace_create(EntryName(standard_workspace_name))
+    wid_personal = await alice_user_fs.workspace_create(EntryName(personal_workspace_name))
+    await alice_user_fs.sync()
+
+    def check_mountpoint_paths(standard_mountpoint_path, personal_mountpoint_path):
+        # Both workspaces should be mounted on standard_base_mountpoint
+        assert (
+            standard_mountpoint_path
+            == (standard_base_mountpoint / standard_workspace_name).absolute()
+        )
+        assert (
+            personal_mountpoint_path
+            == (standard_base_mountpoint / personal_workspace_name).absolute()
+        )
+
+    # In the following scenarios, workspaces should be mounted on
+    # the standard base mountpoint
+
+    async with mountpoint_manager_factory(
+        alice_user_fs,
+        alice_user_fs.event_bus,
+        standard_base_mountpoint,
+        mountpoint_in_directory=True,
+        # Mount without specifying a path for personal workspaces
+        # personal_workspace_base_path=personal_base_mountpoint,
```

Comment:
dead code ?

Reply:
It was just to emphasize that this parameter is not being used. But I agree that it is not very clear. I've removed the dead code.